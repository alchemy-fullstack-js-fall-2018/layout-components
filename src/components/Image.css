.image {
display: flex;
flex-direction: row;
flex-wrap: wrap;
align-items: center;
justify-self: center;
justify-content: space-around;
img {
  margin: 10px;
  width: 350px;
  height: auto;
}
}