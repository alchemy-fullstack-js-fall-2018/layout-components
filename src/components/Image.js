import Component from './Component';
import styles from './Image.css';

export default class Image extends Component {
  render() {
    const images = this.props.images.map(img => {
      return `<img src="./src/assets/${img}"></img>`;
    }).join(' ');

    return `
        <div class="${styles.image}">
          ${images}
        </div>
       `;
  }
}
