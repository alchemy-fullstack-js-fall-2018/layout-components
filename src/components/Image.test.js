import Image from './Image';

describe('Image Component', () => {
  it('has finalized scaffolding/styling', () => {
    const image = new Image({
      images: ['dummy.jpg']
    });

    expect(image.render()).toMatchSnapshot();
  });
});
