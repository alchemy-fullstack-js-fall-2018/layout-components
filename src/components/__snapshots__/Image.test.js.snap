// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component has finalized scaffolding/styling 1`] = `
"
        <div class=\\"image\\">
          <img src=\\"./src/assets/dummy.jpg\\"></img>
        </div>
       "
`;
